crates/ide-assists/src/handlers/generate_enum_variant.rs

```diff
@@ -0,0 +1,195 @@
+use hir::HasSource;
+use ide_db::assists::{AssistId, AssistKind};
+use syntax::{
+    ast::{self, edit::IndentLevel},
+    AstNode, TextSize,
+};
+
+use crate::assist_context::{AssistContext, Assists};
+
+// Assist: generate_enum_variant
+//
+// Adds a variant to an enum.
+//
+// ```
+// enum Countries {
+//     Ghana,
+// }
+//
+// fn main() {
+//     let country = Countries::Lesotho$0;
+// }
+// ```
+// ->
+// ```
+// enum Countries {
+//     Ghana,
+//     Lesotho,
+// }
+//
+// fn main() {
+//     let country = Countries::Lesotho;
+// }
+// ```
+pub(crate) fn generate_enum_variant(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let path_expr: ast::PathExpr = ctx.find_node_at_offset()?;
+    let path = path_expr.path()?;
+
+    if ctx.sema.resolve_path(&path).is_some() {
+        // No need to generate anything if the path resolves
+        return None;
+    }
+
+    let name_ref = path.segment()?.name_ref()?;
+    if name_ref.text().as_str().chars().next()?.is_ascii_lowercase() {
```

Comment:
```suggestion
    if name_ref.text().as_str().starts_with(char::is_lowercase) {
```
Similar can be done for the uppercase check in `generate_function`. Going with the non-ascii version probably makes more more sense? (though I don't really know of anyone NOT using the latin alphabet set for identifiers)

Reply:
Done in 796c4d8a10359bc8add6e5acf952c87f4354189b